external/skiboot/libstb/secvar/crypto/crypto-openssl.c

```diff
@@ -667,7 +665,9 @@ int crypto_md_generate_hash(const unsigned char *data, size_t size,
 	if (verbose) {
 		printf("Hash generation successful, %s: ",
 		       OBJ_nid2sn(hashFunct));
-		printHex(*outHash, *outHashSize);
+		for (int i = 0; i < *outHashSize; i++)
+			printf("/%02x", (*outHash)[i]);
```

Comment:
(I realise that you're copying over your existing behaviour, but...) Wouldn't it be more conventional to use `:` as a separator rather than `/`?

Also, perhaps to avoid the separator preceding the first hash value.
```
printf("%02x", (*outHash)[0]);
for (int i = 1; i < *outHashSize; i++) // (also, does our C spec variant allow loop counters to be declared here?)
    printf(":%02x", ...)
```

Reply:
We're not adhering to a particularly restrictive C spec (that I'm aware of), but I would prefer not having initializations inside for-loops.

Reply:
This may be worthy of an entire new PR. All for loops in secvarctl use internal counter initializations. In terms of C spec, we use gnu99 which is C99 with gnu extensions. It appears that C99 allows for declarations in a for loop. Let me know where we want to go with this.

Reply:
Considering this is a skiboot file, we should adhere to their usage. A quick grep shows at least a few instances of initializing in the loop, but ultimately it will be up to the current maintainer.

I value consistency more than my personal gripes though, so if these are already consistently used in secvarctl, I'm fine with whichever.